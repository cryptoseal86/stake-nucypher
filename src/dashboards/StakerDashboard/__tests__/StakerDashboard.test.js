import React from 'react';
import StakerDashboard from '../StakerDashboard';
import { render, fireEvent, wait } from '@testing-library/react';
import Web3Initilizer from '../../../web3Initializer';
jest.mock('../../../web3Initializer');

describe('<StakerDashboard />', () => {

  const originalDateNow = Date.now;
  const originalDateGetTime = Date.prototype.getTime;
  let escrowContract;
  beforeAll(() => {
    Date.now = () => 1582581711708;
    Date.prototype.getTime = () => 1582581711708; // eslint-disable-line
    escrowContract = Web3Initilizer.getContractInstance();
  });

  afterAll(() => {
    Date.now = originalDateNow;
    Date.prototype.getTime = originalDateGetTime; // eslint-disable-line
    jest.restoreMocks();
  });

  it('should render correctly', async () => {
    await wait(() => {
      const { asFragment } = render(<StakerDashboard />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it('should call contract method on setting restake', async () => {
    await wait(() => {
      const { getByTestId } = render(<StakerDashboard />);
      fireEvent.click(getByTestId('restaking-toggle'));
    });
    expect(escrowContract.methods.setReStake).toBeCalled();
  });

  it('should call contract method on setting wind down', async () => {
    await wait(() => {
      const { getByTestId } = render(<StakerDashboard />);
      fireEvent.click(getByTestId('winddown-toggle'));
    });
    expect(escrowContract.methods.setWindDown).toBeCalled();
  });

  it('should show modal on add stake button click', async () => {
    await wait(() => {
      const { getByText, asFragment } = render(<StakerDashboard />);
      fireEvent.click(getByText('Add stake'));
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it('should show modal on set worker button click', async () => {
    await wait(() => {
      const { getByText, asFragment } = render(<StakerDashboard />);
      fireEvent.click(getByText('Change worker'));
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it('should call contract method on detach worker', async () => {
    await wait(() => {
      const { getByText } = render(<StakerDashboard />);
      fireEvent.click(getByText('Detach worker'));
    });
    expect(escrowContract.methods.setWorker).toBeCalled();
  });
});
