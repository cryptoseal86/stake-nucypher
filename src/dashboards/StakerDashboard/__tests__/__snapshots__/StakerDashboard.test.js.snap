// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StakerDashboard /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="panel row"
  >
    <div
      class="col"
    >
      <h2
        class="text-center"
      >
        Welcome back, 
      </h2>
      <div
        class="mt-5 row"
      >
        <div
          class="col"
        >
          <p
            class="h6 text-center"
          >
            Current period
          </p>
          <p
            class="h4 text-center"
          >
            18945
          </p>
        </div>
        <div
          class="col"
        >
          <p
            class="h6 text-center"
          >
            Next period in
          </p>
          <p
            class="h4 text-center"
          >
            0 minutes
          </p>
        </div>
      </div>
      <div
        class="mt-5 tokens-row row"
      >
        <div
          class="col"
        >
          <p
            class="h6 text-center"
          >
            Tokens staked
          </p>
          <p
            class="h4 text-center"
          >
             
            <br />
            NU
          </p>
          <div
            class="action d-flex justify-content-center"
          >
            <button
              class="button-action mt-2 btn btn-primary"
              type="button"
            >
              Add stake
            </button>
          </div>
        </div>
        <div
          class="col"
        >
          <p
            class="h6 text-center"
          >
            Total rewards
          </p>
          <p
            class="h4 text-center"
          >
             
            <br />
            NU
          </p>
        </div>
        <div
          class="col"
        >
          <p
            class="h6 text-center"
          >
            Available for withdrawal
          </p>
          <p
            class="h4 text-center"
          >
             
            <br />
            NU
          </p>
          <div
            class="action d-flex justify-content-center"
          >
            <button
              class="button-action mt-2 btn btn-secondary"
              data-testid="withdraw-switch-button"
              type="button"
            >
              Withdraw
            </button>
          </div>
        </div>
      </div>
      <div
        class="mt-5 row"
      />
      <div
        class="mt-5 row"
      />
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col"
    />
  </div>
</DocumentFragment>
`;

exports[`<StakerDashboard /> should show modal on add stake button click 1`] = `
<DocumentFragment>
  <div
    class="panel row"
  >
    <div
      class="col"
    >
      <h2
        class="text-center"
      >
        Welcome back, 0x62DB
      </h2>
      <div
        class="mt-5 row"
      >
        <div
          class="col"
        >
          <p
            class="h6 text-center"
          >
            Current period
          </p>
          <p
            class="h4 text-center"
          >
            18945
          </p>
        </div>
        <div
          class="col"
        >
          <p
            class="h6 text-center"
          >
            Next period in
          </p>
          <p
            class="h4 text-center"
          >
            0 minutes
          </p>
        </div>
      </div>
      <div
        class="mt-5 tokens-row row"
      >
        <div
          class="col"
        >
          <p
            class="h6 text-center"
          >
            Tokens staked
          </p>
          <p
            class="h4 text-center"
          >
            179272.6235 
            <br />
            NU
          </p>
          <div
            class="action d-flex justify-content-center"
          >
            <button
              class="button-action mt-2 btn btn-primary"
              type="button"
            >
              Add stake
            </button>
          </div>
        </div>
        <div
          class="col"
        >
          <p
            class="h6 text-center"
          >
            Total rewards
          </p>
          <p
            class="h4 text-center"
          >
             
            <br />
            NU
          </p>
        </div>
        <div
          class="col"
        >
          <p
            class="h6 text-center"
          >
            Available for withdrawal
          </p>
          <p
            class="h4 text-center"
          >
            179272.6235 
            <br />
            NU
          </p>
          <div
            class="action d-flex justify-content-center"
          >
            <button
              class="button-action mt-2 btn btn-secondary"
              data-testid="withdraw-switch-button"
              type="button"
            >
              Withdraw
            </button>
          </div>
        </div>
      </div>
      <div
        class="mt-5 row"
      >
        <div
          class="col-md-6"
        >
          <p
            class="h6 text-center"
          >
            Restaking
          </p>
          <div
            class="h4 text-center"
          >
            <div
              class="react-toggle"
            >
              <div
                class="react-toggle-track"
              >
                <div
                  class="react-toggle-track-check"
                >
                  <svg
                    height="11"
                    viewBox="0 0 14 11"
                    width="14"
                  >
                    <title>
                      switch-check
                    </title>
                    <path
                      d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
                      fill="#fff"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
                <div
                  class="react-toggle-track-x"
                >
                  <svg
                    height="10"
                    viewBox="0 0 10 10"
                    width="10"
                  >
                    <title>
                      switch-x
                    </title>
                    <path
                      d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
                      fill="#fff"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="react-toggle-thumb"
              />
              <input
                class="react-toggle-screenreader-only"
                data-testid="restaking-toggle"
                type="checkbox"
              />
            </div>
          </div>
        </div>
        <div
          class="col-md-6"
        >
          <p
            class="h6 text-center"
          >
            Wind down
          </p>
          <div
            class="h4 text-center"
          >
            <div
              class="react-toggle"
            >
              <div
                class="react-toggle-track"
              >
                <div
                  class="react-toggle-track-check"
                >
                  <svg
                    height="11"
                    viewBox="0 0 14 11"
                    width="14"
                  >
                    <title>
                      switch-check
                    </title>
                    <path
                      d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
                      fill="#fff"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
                <div
                  class="react-toggle-track-x"
                >
                  <svg
                    height="10"
                    viewBox="0 0 10 10"
                    width="10"
                  >
                    <title>
                      switch-x
                    </title>
                    <path
                      d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
                      fill="#fff"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="react-toggle-thumb"
              />
              <input
                class="react-toggle-screenreader-only"
                data-testid="winddown-toggle"
                type="checkbox"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="mt-5 row"
      >
        <div
          class="col"
        >
          <p
            class="h6 text-center"
          >
            Your worker
          </p>
          <p
            class="h4 text-center"
          >
            0x62DB
          </p>
          <div
            class="action d-flex justify-content-center text-center"
          >
            <button
              class="btn btn-secondary"
              type="button"
            >
              Change worker
            </button>
            <button
              class="ml-1 btn btn-secondary"
              type="button"
            >
              Detach worker
            </button>
          </div>
          <div
            class="small text-muted text-center pt-1"
          >
            Warning! Worker change/detach is possible once in 2 periods
          </div>
        </div>
        <div
          class="col"
        >
          <p
            class="h6 text-center"
          >
            Worker activity
          </p>
          <p
            class="h5 text-center"
          >
            <span
              class="text-secondary"
            >
              <span>
                Never confirmed activity
              </span>
            </span>
          </p>
          <div
            class="action"
          />
          <div
            class="action d-flex justify-content-center"
          >
            <button
              class="primary btn btn-primary"
              type="button"
            >
              Follow on Telegram 💬
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <h5
            class="pt-3 pb-3 text-center"
          >
            Your stakes:
          </h5>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div
            class="container"
          >
            <div
              class="table"
            >
              <div
                class="table-row table-header"
              >
                <div
                  class="table-row-column table-row-index"
                >
                  #
                </div>
                <div
                  class="table-row-column table-center"
                >
                  Value
                </div>
                <div
                  class="table-row-column table-center"
                >
                  Remaining
                </div>
                <div
                  class="table-row-column table-center"
                >
                  Enactment
                </div>
                <div
                  class="table-row-column table-center"
                />
              </div>
              <div
                class="table-row"
              >
                <div
                  class="table-row-column table-row-index"
                >
                  0
                </div>
                <div
                  class="table-row-column table-center"
                >
                  64369.1548 NU
                </div>
                <div
                  class="table-row-column table-center"
                >
                  366 periods
                </div>
                <div
                  class="table-row-column table-center"
                >
                  Jan 24, 20 - Feb 24, 21
                </div>
                <div
                  class="table-row-column table-center"
                >
                  <button
                    class="m-1 btn btn-secondary btn-sm"
                    data-testid="divide-button-0"
                    type="button"
                  >
                    Divide
                  </button>
                  <button
                    class="m-1 btn btn-secondary btn-sm"
                    data-testid="prolong-button-0"
                    type="button"
                  >
                    Prolong
                  </button>
                </div>
              </div>
              <div
                class="mt-4 collapse container"
              >
                <div
                  class="row"
                >
                  <div
                    class="col"
                  >
                    <form
                      class=""
                    >
                      <div
                        class="mt-2 form-group row"
                      >
                        <div
                          class="col-sm-3"
                        >
                          <input
                            class="form-control form-control-sm"
                            data-testid="first-substake-value"
                            name="firstSubstakeValue"
                            type="text"
                            value="32184.57741283639504424"
                          />
                          <div
                            class="feedback-placeholder"
                          >
                            <div
                              class="invalid-feedback"
                              style="display: none;"
                            />
                          </div>
                        </div>
                        <div
                          class="d-flex justify-content-center col-sm-6"
                          style="padding-top: 8px;"
                        >
                          <div
                            class="rc-slider"
                          >
                            <div
                              class="rc-slider-rail"
                            />
                            <div
                              class="rc-slider-track"
                              style="left: 0%; width: 50%;"
                            />
                            <div
                              class="rc-slider-step"
                            />
                            <div
                              aria-disabled="false"
                              aria-valuemax="100"
                              aria-valuemin="0"
                              aria-valuenow="50"
                              class="rc-slider-handle"
                              role="slider"
                              style="left: 50%; transform: translateX(-50%);"
                              tabindex="0"
                            />
                            <div
                              class="rc-slider-mark"
                            />
                          </div>
                        </div>
                        <div
                          class="col-sm-3"
                        >
                          <input
                            class="form-control form-control-sm"
                            data-testid="second-substake-value"
                            name="secondSubstakeValue"
                            type="text"
                            value="32184.577412836395044239"
                          />
                          <div
                            class="feedback-placeholder"
                          >
                            <div
                              class="invalid-feedback"
                              style="display: none;"
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class="form-group row"
                      >
                        <label
                          class="form-label col-form-label col-md-2 offset-md-2"
                        >
                          Extend for
                        </label>
                        <div
                          class="col-md-4"
                        >
                          <input
                            class="form-control"
                            name="stakeDuration"
                            type="number"
                            value="1"
                          />
                          <div
                            class="feedback-placeholder"
                          >
                            <div
                              class="invalid-feedback"
                              style="display: none;"
                            />
                          </div>
                        </div>
                        <label
                          class="form-label col-form-label col-md-2"
                        >
                          periods
                        </label>
                      </div>
                      <div
                        class="d-flex justify-content-center"
                      >
                        <button
                          class="btn btn-primary"
                          type="submit"
                        >
                          Divide
                        </button>
                      </div>
                    </form>
                  </div>
                </div>
              </div>
              <div
                class="mt-4 collapse container"
              >
                <div
                  class="row"
                >
                  <div
                    class="col"
                  >
                    <form
                      class=""
                    >
                      <div
                        class="form-group row"
                      >
                        <label
                          class="form-label col-form-label col-md-2 offset-md-2"
                          for="substake-prolong-duration"
                        >
                          Extend for
                        </label>
                        <div
                          class="col-md-4"
                        >
                          <input
                            class="form-control"
                            id="substake-prolong-duration"
                            name="prolongDurationValue"
                            type="number"
                            value="1"
                          />
                          <div
                            class="feedback-placeholder"
                          >
                            <div
                              class="invalid-feedback"
                              style="display: none;"
                            />
                          </div>
                        </div>
                        <label
                          class="form-label col-form-label col-md-4"
                        >
                          periods
                        </label>
                      </div>
                      <div
                        class="d-flex justify-content-center"
                      >
                        <button
                          class="btn btn-primary"
                          type="submit"
                        >
                          Prolong
                        </button>
                      </div>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<StakerDashboard /> should show modal on set worker button click 1`] = `
<DocumentFragment>
  <div
    class="panel row"
  >
    <div
      class="col"
    >
      <h2
        class="text-center"
      >
        Welcome back, 0x62DB
      </h2>
      <div
        class="mt-5 row"
      >
        <div
          class="col"
        >
          <p
            class="h6 text-center"
          >
            Current period
          </p>
          <p
            class="h4 text-center"
          >
            18945
          </p>
        </div>
        <div
          class="col"
        >
          <p
            class="h6 text-center"
          >
            Next period in
          </p>
          <p
            class="h4 text-center"
          >
            0 minutes
          </p>
        </div>
      </div>
      <div
        class="mt-5 tokens-row row"
      >
        <div
          class="col"
        >
          <p
            class="h6 text-center"
          >
            Tokens staked
          </p>
          <p
            class="h4 text-center"
          >
            179272.6235 
            <br />
            NU
          </p>
          <div
            class="action d-flex justify-content-center"
          >
            <button
              class="button-action mt-2 btn btn-primary"
              type="button"
            >
              Add stake
            </button>
          </div>
        </div>
        <div
          class="col"
        >
          <p
            class="h6 text-center"
          >
            Total rewards
          </p>
          <p
            class="h4 text-center"
          >
             
            <br />
            NU
          </p>
        </div>
        <div
          class="col"
        >
          <p
            class="h6 text-center"
          >
            Available for withdrawal
          </p>
          <p
            class="h4 text-center"
          >
            179272.6235 
            <br />
            NU
          </p>
          <div
            class="action d-flex justify-content-center"
          >
            <button
              class="button-action mt-2 btn btn-secondary"
              data-testid="withdraw-switch-button"
              type="button"
            >
              Withdraw
            </button>
          </div>
        </div>
      </div>
      <div
        class="mt-5 row"
      >
        <div
          class="col-md-6"
        >
          <p
            class="h6 text-center"
          >
            Restaking
          </p>
          <div
            class="h4 text-center"
          >
            <div
              class="react-toggle"
            >
              <div
                class="react-toggle-track"
              >
                <div
                  class="react-toggle-track-check"
                >
                  <svg
                    height="11"
                    viewBox="0 0 14 11"
                    width="14"
                  >
                    <title>
                      switch-check
                    </title>
                    <path
                      d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
                      fill="#fff"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
                <div
                  class="react-toggle-track-x"
                >
                  <svg
                    height="10"
                    viewBox="0 0 10 10"
                    width="10"
                  >
                    <title>
                      switch-x
                    </title>
                    <path
                      d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
                      fill="#fff"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="react-toggle-thumb"
              />
              <input
                class="react-toggle-screenreader-only"
                data-testid="restaking-toggle"
                type="checkbox"
              />
            </div>
          </div>
        </div>
        <div
          class="col-md-6"
        >
          <p
            class="h6 text-center"
          >
            Wind down
          </p>
          <div
            class="h4 text-center"
          >
            <div
              class="react-toggle"
            >
              <div
                class="react-toggle-track"
              >
                <div
                  class="react-toggle-track-check"
                >
                  <svg
                    height="11"
                    viewBox="0 0 14 11"
                    width="14"
                  >
                    <title>
                      switch-check
                    </title>
                    <path
                      d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
                      fill="#fff"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
                <div
                  class="react-toggle-track-x"
                >
                  <svg
                    height="10"
                    viewBox="0 0 10 10"
                    width="10"
                  >
                    <title>
                      switch-x
                    </title>
                    <path
                      d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
                      fill="#fff"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="react-toggle-thumb"
              />
              <input
                class="react-toggle-screenreader-only"
                data-testid="winddown-toggle"
                type="checkbox"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="mt-5 row"
      >
        <div
          class="col"
        >
          <p
            class="h6 text-center"
          >
            Your worker
          </p>
          <p
            class="h4 text-center"
          >
            0x62DB
          </p>
          <div
            class="action d-flex justify-content-center text-center"
          >
            <button
              class="btn btn-secondary"
              type="button"
            >
              Change worker
            </button>
            <button
              class="ml-1 btn btn-secondary"
              type="button"
            >
              Detach worker
            </button>
          </div>
          <div
            class="small text-muted text-center pt-1"
          >
            Warning! Worker change/detach is possible once in 2 periods
          </div>
        </div>
        <div
          class="col"
        >
          <p
            class="h6 text-center"
          >
            Worker activity
          </p>
          <p
            class="h5 text-center"
          >
            <span
              class="text-secondary"
            >
              <span>
                Never confirmed activity
              </span>
            </span>
          </p>
          <div
            class="action"
          />
          <div
            class="action d-flex justify-content-center"
          >
            <button
              class="primary btn btn-primary"
              type="button"
            >
              Follow on Telegram 💬
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <h5
            class="pt-3 pb-3 text-center"
          >
            Your stakes:
          </h5>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div
            class="container"
          >
            <div
              class="table"
            >
              <div
                class="table-row table-header"
              >
                <div
                  class="table-row-column table-row-index"
                >
                  #
                </div>
                <div
                  class="table-row-column table-center"
                >
                  Value
                </div>
                <div
                  class="table-row-column table-center"
                >
                  Remaining
                </div>
                <div
                  class="table-row-column table-center"
                >
                  Enactment
                </div>
                <div
                  class="table-row-column table-center"
                />
              </div>
              <div
                class="table-row"
              >
                <div
                  class="table-row-column table-row-index"
                >
                  0
                </div>
                <div
                  class="table-row-column table-center"
                >
                  64369.1548 NU
                </div>
                <div
                  class="table-row-column table-center"
                >
                  366 periods
                </div>
                <div
                  class="table-row-column table-center"
                >
                  Jan 24, 20 - Feb 24, 21
                </div>
                <div
                  class="table-row-column table-center"
                >
                  <button
                    class="m-1 btn btn-secondary btn-sm"
                    data-testid="divide-button-0"
                    type="button"
                  >
                    Divide
                  </button>
                  <button
                    class="m-1 btn btn-secondary btn-sm"
                    data-testid="prolong-button-0"
                    type="button"
                  >
                    Prolong
                  </button>
                </div>
              </div>
              <div
                class="mt-4 collapse container"
              >
                <div
                  class="row"
                >
                  <div
                    class="col"
                  >
                    <form
                      class=""
                    >
                      <div
                        class="mt-2 form-group row"
                      >
                        <div
                          class="col-sm-3"
                        >
                          <input
                            class="form-control form-control-sm"
                            data-testid="first-substake-value"
                            name="firstSubstakeValue"
                            type="text"
                            value="32184.57741283639504424"
                          />
                          <div
                            class="feedback-placeholder"
                          >
                            <div
                              class="invalid-feedback"
                              style="display: none;"
                            />
                          </div>
                        </div>
                        <div
                          class="d-flex justify-content-center col-sm-6"
                          style="padding-top: 8px;"
                        >
                          <div
                            class="rc-slider"
                          >
                            <div
                              class="rc-slider-rail"
                            />
                            <div
                              class="rc-slider-track"
                              style="left: 0%; width: 50%;"
                            />
                            <div
                              class="rc-slider-step"
                            />
                            <div
                              aria-disabled="false"
                              aria-valuemax="100"
                              aria-valuemin="0"
                              aria-valuenow="50"
                              class="rc-slider-handle"
                              role="slider"
                              style="left: 50%; transform: translateX(-50%);"
                              tabindex="0"
                            />
                            <div
                              class="rc-slider-mark"
                            />
                          </div>
                        </div>
                        <div
                          class="col-sm-3"
                        >
                          <input
                            class="form-control form-control-sm"
                            data-testid="second-substake-value"
                            name="secondSubstakeValue"
                            type="text"
                            value="32184.577412836395044239"
                          />
                          <div
                            class="feedback-placeholder"
                          >
                            <div
                              class="invalid-feedback"
                              style="display: none;"
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class="form-group row"
                      >
                        <label
                          class="form-label col-form-label col-md-2 offset-md-2"
                        >
                          Extend for
                        </label>
                        <div
                          class="col-md-4"
                        >
                          <input
                            class="form-control"
                            name="stakeDuration"
                            type="number"
                            value="1"
                          />
                          <div
                            class="feedback-placeholder"
                          >
                            <div
                              class="invalid-feedback"
                              style="display: none;"
                            />
                          </div>
                        </div>
                        <label
                          class="form-label col-form-label col-md-2"
                        >
                          periods
                        </label>
                      </div>
                      <div
                        class="d-flex justify-content-center"
                      >
                        <button
                          class="btn btn-primary"
                          type="submit"
                        >
                          Divide
                        </button>
                      </div>
                    </form>
                  </div>
                </div>
              </div>
              <div
                class="mt-4 collapse container"
              >
                <div
                  class="row"
                >
                  <div
                    class="col"
                  >
                    <form
                      class=""
                    >
                      <div
                        class="form-group row"
                      >
                        <label
                          class="form-label col-form-label col-md-2 offset-md-2"
                          for="substake-prolong-duration"
                        >
                          Extend for
                        </label>
                        <div
                          class="col-md-4"
                        >
                          <input
                            class="form-control"
                            id="substake-prolong-duration"
                            name="prolongDurationValue"
                            type="number"
                            value="1"
                          />
                          <div
                            class="feedback-placeholder"
                          >
                            <div
                              class="invalid-feedback"
                              style="display: none;"
                            />
                          </div>
                        </div>
                        <label
                          class="form-label col-form-label col-md-4"
                        >
                          periods
                        </label>
                      </div>
                      <div
                        class="d-flex justify-content-center"
                      >
                        <button
                          class="btn btn-primary"
                          type="submit"
                        >
                          Prolong
                        </button>
                      </div>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
